File: 09.py
with open('09.txt') as f:
    s = [c for c in f.read().strip()]

garbage = False
score = 0
level = 0
gscore = 0
while s:
    c = s.pop(0)
    if s and c == '!':
        c = s.pop(0)
    elif not garbage and c == '<':
        garbage = True
    elif garbage and c == '>':
        garbage = False
    elif not garbage and c != ',':
        if c == '{':
            level += 1
        else:
            score += level
            level -= 1
    elif garbage:
        gscore += 1

print('Part 1:',score)
print('Part 2:',gscore)

File: 09.jl
open("09.txt") do f
    global l = readline(f)
end

function solver(chars)
    cs = collect(l)
    garbage = false
    score = 0
    level = 0
    gscore = 0
    while length(cs) > 0
        c = popfirst!(cs)
        if length(cs)>0 && c == '!'
            c = popfirst!(cs)
        elseif ! garbage && c == '<'
            garbage = true
        elseif garbage && c == '>'
            garbage = false
        elseif ! garbage && c != ','
            if c == '{'
                level += 1
            else
                score += level
                level -= 1
            end
        elseif garbage
            gscore += 1
        end
    end
    return score, gscore
end

score, gscore = solver(l)
println("Part 1: ", score)
println("Part 2: ", gscore)
